### ISSUE TITLE
Error while promote with bitbucket server

### ISSUE THREAD

**Issue by ysaakpr** [jenkins-x/jx#857]
+ jx promote -b --env staging --timeout 1h --version 0.0.48

Promoting app tools-frontend version 0.0.48 to namespace staging

Using pipeline: livspaceeng/tools-frontend/master build: #10

Cloning into '/home/jenkins/.jx/environments/livspace****/environment-jx-staging'...

Found remote branch names master, HEAD -> origin/master, master, promote-tools-frontend-0.0.40, promote-tools-frontend-0.0.44, promote-tools-frontend-0.0.45, promote-tools-frontend-0.0.46, promote-tools-frontend-0.0.47

Switched to branch 'promote-tools-frontend-0.0.48'

[promote-tools-frontend-0.0.48 f3c0ea0] Promote tools-frontend to version 0.0.48

 1 file changed, 1 insertion(+), 1 deletion(-)

remote: 

remote: Create pull request for promote-tools-frontend-0.0.48:        

remote:   https://bitbucket.org/livspace****/environment-jx-staging/pull-requests/new?source=promote-tools-frontend-0.0.48&t=1        

remote: 

To https://bitbucket.org/livspace****/environment-jx-staging.git

 * [new branch]      HEAD -> promote-tools-frontend-0.0.48

Created Pull Request: https://bitbucket.org/livspace****/environment-jx-staging/pull-requests/6

Using pipeline: livspaceeng/tools-frontend/master build: #10

Pull Request https://bitbucket.org/livspace****/environment-jx-staging/pull-requests/6 is merged at sha 3d465a9ff5f2

WARNING: Failed to query merge status of repo livspace****/livspace****/environment-jx-staging with merge sha 3d465a9ff5f2 due to: Status: 404 Not Found, Body: {"type": "error", "error": {"message": "Resource not found", "detail": "There is no API hosted at this URL.\n\nFor information about our API's, please refer to the documentation at: https://developer.atlassian.com/bitbucket/api/2/reference/"}}

The build is hung at this state but i can see the deployment is happened

**Comment by wbrefvem** [comment]
There are still some bugs around PRs on Bitbucket Cloud that seem to show up in `jx promote`. Hopefully they'll be worked out soon.

**Comment by ysaakpr** [comment]
Error got fixed, but still the build is hung with no status update
```
Pull Request https://bitbucket.org/livspace****/environment-jx-beta/pull-requests/5 is merged at sha ac8fd5b0aafd
Merge commit has not yet any statuses on repo livspace****/livspace****/environment-jx-beta merge sha ac8fd5b0aafd
```

**Comment by wbrefvem** [comment]
@ysaakpr yeah, there's still an issue with the status ID. At the moment, we use the status key from Bitbucket, but it doesn't work because the Bitbucket Branch Source Plugin sets the key to an md5 hash of the Jenkins job name, which means it's 128-bit and won't fit into an `int64`. So we probably need to make the status ID a string so that we can accommodate status IDs of arbitrary length.

**Comment by wbrefvem** [comment]
@ysaakpr Your issue should be resolved if you update to the latest release. Please give it a try and let us know if you're continuing to have issues.

**Comment by ysaakpr** [comment]
I dont know whether this issue fixed or not...becase now i am getting different error with respect environment 
```
[_environment-jx-beta_master-G4D33OX4LAGVVF63566PZYCO4ZQ6UKQ2OTZBMP45YAJXRNUM5EFA] Running shell script
+ make install
helm upgrade "beta" "env" --install --namespace "beta" --debug
[debug] Created tunnel using local port: '38159'

[debug] SERVER: "127.0.0.1:38159"
Error: UPGRADE FAILED: "beta" has no deployed releases
make: *** [install] Error 1
script returned exit code 2
```

**Comment by ysaakpr** [comment]
Still hung at Merge commit has not yet any statuses on repo even though the deployment of environment happened successfully
```
Switched to branch 'promote-tools-frontend-0.0.60'
[promote-tools-frontend-0.0.60 23b998b] Promote tools-frontend to version 0.0.60
 1 file changed, 8 insertions(+), 5 deletions(-)
remote: 
remote: Create pull request for promote-tools-frontend-0.0.60:        
remote:   https://bitbucket.org/livspace****/environment-jx-beta/pull-requests/new?source=promote-tools-frontend-0.0.60&t=1        
remote: 
To https://bitbucket.org/livspace****/environment-jx-beta.git
 * [new branch]      HEAD -> promote-tools-frontend-0.0.60
Created Pull Request: https://bitbucket.org/livspace****/environment-jx-beta/pull-requests/1

Using pipeline: livspaceeng/tools-frontend/master build: #2
Pull Request https://bitbucket.org/livspace****/environment-jx-beta/pull-requests/1 is merged at sha 1e73783f6aa3

Merge commit has not yet any statuses on repo livspace****/livspace****/environment-jx-beta merge sha 1e73783f6aa3
```

**Comment by ysaakpr** [comment]
All my builds keep waiting for the merge to happen but even through the merge is done and promote is successful, builds are not able to complete the promotion. Is there any progress on this.. Is there anything that i am doing wrong with my bitbucket repo here

**Comment by wbrefvem** [comment]
@ysaakpr It's hard to say without knowing a lot more detail. How did you install Jenkins X? Did you do `jx create cluster` or did you `jx install` on an existing cluster?

**Comment by LinuxSuRen** [comment]
I meet the same problem. Do you mean you manually set 'Jenkins Location'? I already set this, but still get hung.
